packages/svelte/scripts/process-messages/index.js

```diff
@@ -98,55 +99,18 @@ function run() {
 			.replace(/\r\n/g, '\n');
 
 		/**
-		 * @type {Array<{
-		 * 	type: string;
-		 * 	value: string;
-		 * 	start: number;
-		 * 	end: number
-		 * }>}
+		 * @type {any[]}
 		 */
 		const comments = [];
 
 		let ast = acorn.parse(source, {
 			ecmaVersion: 'latest',
 			sourceType: 'module',
-			onComment: (block, value, start, end) => {
-				if (block && /\n/.test(value)) {
-					let a = start;
-					while (a > 0 && source[a - 1] !== '\n') a -= 1;
-
-					let b = a;
-					while (/[ \t]/.test(source[b])) b += 1;
-
-					const indentation = source.slice(a, b);
-					value = value.replace(new RegExp(`^${indentation}`, 'gm'), '');
-				}
-
-				comments.push({ type: block ? 'Block' : 'Line', value, start, end });
-			}
+			locations: true,
+			onComment: comments
```

Comment:
this seems... weird

Reply:
just a quirk of the acorn API, you can pass a callback or an array. in the latter case the array gets populated